iceoryx_posh/source/runtime/shared_memory_user.cpp

```diff
@@ -73,7 +75,23 @@ void SharedMemoryUser::openDataSegments(const uint64_t segmentId,
     {
         auto accessMode = segment.m_isWritable ? AccessMode::READ_WRITE : AccessMode::READ_ONLY;
         PosixSharedMemoryObjectBuilder()
-            .name(segment.m_sharedMemoryName)
+            .name([&segment] {
+                using ShmName_t = detail::PosixSharedMemory::Name_t;
+                ShmName_t shmName = iceoryxResourcePrefix(roudi::DEFAULT_UNIQUE_ROUDI_ID,
+                                                          "p"); // use an additional 'p' to prevent creating a payload
+                                                                // segment with the same name as the management segment
+                if (shmName.size() + segment.m_sharedMemoryName.size() > ShmName_t::capacity())
```

Comment:
I think a valid `ShmName` is always a `FileName` and this class already provides methods to add/concat/modify the content in a semantic correct way. Should we maybe use it, instead of `ShmName` - it already does all the error handling for us, is nicely tested etc. etc.

Reply:
If you mean to replace `detail::PosixSharedMemory::Name_t` with `FileName` I would see that as out of scope for this PR. I can create an issue for this though.

For the specific code in general it does not really matter since the only thing which is different to before is the additional prefix and this is fully in our control and therefore no invalid characters will be added.

Reply:
Added to #2018